File: Euclid/Messages/Incoming/ADDSTRIPITEM.cs
using Euclid.Game;
using Euclid.Messages.Outgoing;
using Euclid.Network.Streams;
using Euclid.Util.Extensions;
using Euclid.Util;

namespace Euclid.Messages.Incoming
{
    class ADDSTRIPITEM : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            if (player.RoomEntity.Room == null)
                return;

            Room room = player.RoomEntity.Room;
            var placementData = request.Content.Split(' ');

            if (!placementData[2].IsNumeric())
                return;

            int itemId = int.Parse(placementData[2]);
            Item item = room.ItemManager.GetItem(itemId);

            if (item == null)
                return;

            if (room == null || !room.IsOwner(player.Details.Id))
            {
                return;
            }

            room.FurnitureManager.RemoveItem(item, player);

            player.Inventory.AddItem(item);
            player.Inventory.TurnPage("last");
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/ADDSTRIPITEM.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.item.Item;
import euclid.game.room.Room;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;
import euclid.util.extensions.StringExtensions;

public class ADDSTRIPITEM implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        if (player.getRoomEntity().getRoom() == null) {
            return;
        }

        Room room = player.getRoomEntity().getRoom();
        String[] placementData = request.getContent().split(" ");

        if (placementData.length < 3 || !StringExtensions.isNumeric(placementData[2])) {
            return;
        }

        int itemId = Integer.parseInt(placementData[2]);
        Item item = room.getItemManager().getItem(itemId);

        if (item == null) {
            return;
        }

        if (room == null || !room.isOwner(player.getDetails().getId())) {
            return;
        }

        room.getFurnitureManager().removeItem(item, player);

        player.getInventory().addItem(item);
        player.getInventory().turnPage("last");
    }
}
